source/macos/AdaptiveCards/AdaptiveCards/Renderers/ChoiceSetInputRenderer.swift

```diff
@@ -0,0 +1,93 @@
+            }
+        }
+        // display multi-select check-boxes
+        return choiceSetRenderInternal(choiceSetInput: choiceSetInput, with: hostConfig, style: style)
+    }
+    
+    func choiceSetRenderInternal(choiceSetInput: ACSChoiceSetInput, with hostConfig: ACSHostConfig, style: ACSContainerStyle) -> NSView {
+        // Parse input default values for multi-select
+        let defaultParsedValues = parseChoiceSetInputDefaultValues(value: choiceSetInput.getValue() ?? "")
+        let isMultiSelect = choiceSetInput.getIsMultiSelect()
+        let view = ACRChoiceSetView()
+        view.isRadioGroup = !isMultiSelect
+        view.wrap = choiceSetInput.getWrap()
+        for choice in choiceSetInput.getChoices() {
+            let title = choice.getTitle() ?? ""
+            let attributedString = getAttributedString(title: title, with: hostConfig, style: style)
+            let choiceButton = view.setupButton(attributedString: attributedString, value: choice.getValue())
+            if isMultiSelect {
+                // checkbox
+                choiceButton.type = .switch
+            } else {
+                // radio box
+                choiceButton.type = .radio
+            }
+            choiceButton.backgroundColor = hostConfig.getBackgroundColor(for: style) ?? .clear
+            if defaultParsedValues.contains(choice.getValue() ?? "") {
+                choiceButton.state = .on
+                view.previousButton = choiceButton
+            }
+            view.addChoiceButton(choiceButton)
+        }
+        return view
+    }
+    
+    func parseChoiceSetInputDefaultValues(value: String) -> [String] {
+        let parsedValues = value.components(separatedBy: ",")
+        return parsedValues
+    }
+}
+// MARK: Extension
+extension ChoiceSetInputRenderer {
+    func choiceSetCompactRenderInternal (choiceSetInput: ACSChoiceSetInput, with hostConfig: ACSHostConfig, style: ACSContainerStyle) -> NSView {
+        // compact button renderer
+        let choiceSetFieldCompactView = ACRChoiceSetCompactView()
+        choiceSetFieldCompactView.autoenablesItems = false
+        var index = 0
+        if choiceSetInput.getPlaceholder() != "" {
+            choiceSetFieldCompactView.addItem(withTitle: choiceSetInput.getPlaceholder() ?? "")
+            if let menuItem = choiceSetFieldCompactView.item(at: 0) {
+                menuItem.isEnabled = false
+            }
+            index += 1
+        }
+        for choice in choiceSetInput.getChoices() {
+            let title = choice.getTitle() ?? ""
+            choiceSetFieldCompactView.addItem(withTitle: "")
+            let item = choiceSetFieldCompactView.item(at: index)
+            item?.title = title
+            // item?.attributedTitle = getAttributedString(title: title, with: hostConfig, style: style, wrap: choiceSetInput.getWrap())
+            if choiceSetInput.getValue() == choice.getValue() {
```

Comment:
Can you explain the logic here?

Reply:
yes, actually NSpopupbutton has the property that, it directly doesnot allow elements with same title, we have to override it, otherwise for same title, it removes the previous element